Libraries/JavaScriptAppEngine/System/JSTimers/JSTimersExecution.js

```diff
@@ -70,6 +73,12 @@ var JSTimersExecution = {
       } else if (type === JSTimersExecution.Type.requestAnimationFrame) {
         var currentTime = performanceNow();
         callback(currentTime);
+      } else if (type === JSTimersExecution.Type.requestIdleCallback) {
+        callback({
+          timeRemaining: function() {
+            return Math.max(0, 17 - (performanceNow() - frameTime));
```

Comment:
Just want to chime in and suggest adding a TODO for now if synchronous JS execution isn't yet figured out. For example, if we rely on synchronous JS, the Chrome debugger won't work and therefore you can't debug rIC code. (This limitation seems OK for more obscure libraries, but for a core API like rIC I think we want it to be debuggable.)